include/lsst/afw/geom/detail/frameSetUtils.h

```diff
@@ -0,0 +1,113 @@
+// -*- lsst-c++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2017 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_GEOM_FRAMESETUTILS_H
+#define LSST_AFW_GEOM_FRAMESETUTILS_H
+
+#include <memory>
+#include <vector>
+
+#include "astshim.h"
+#include "ndarray.h"
+
+#include "lsst/afw/geom/Endpoint.h"
+#include "lsst/afw/geom/Transform.h"
+#include "lsst/daf/base/PropertyList.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+namespace detail {
+
+/**
+Get the specified frame of a FrameSet as a SkyFrame
+
+@param[in] frameSet  FrameSet from which to get the SkyFrame
+@param[in] index  Index of SkyFrame.
+        This value should lie in the range 1 to the number of frames in the FrameSet
+        (as given by getNframe). A value of FrameSet::Base or FrameSet::CURRENT
+        may be given to specify the base Frame or the current Frame, respectively.
+@param[in] copy  If true return a deep copy of the SkyFrame, else a shallow copy
+*/
+std::shared_ptr<ast::SkyFrame> getSkyFrame(ast::FrameSet const& frameSet, int index, bool copy);
+
+/**
+Make FITS metadata for a pure tangent WCS
+
+@param[in] crpix  Center of projection on the CCD using the LSST convention:
+                        0, 0 is the lower left pixel of the image
+@param[in] crval  Center of projection on the sky
+@param[in] cdMatrix  CD matrix where element (i-1, j-1) corresponds to FITS keyword CDi_j
+                        and i, j have range [1, 2]
+*/
+std::shared_ptr<daf::base::PropertyList> makeTanWcsMetadata(Point2D const& crpix, SpherePoint const& crval,
+                                                            Eigen::Matrix2d const& cdMatrix);
```

Comment:
> WCS meaning solely "pixel to sky" is a real optical imaging bias that we'd benefit from dropping.

That sounds like an argument for calling this "CartesianToCelestialTransform", not "SkyWcs", which also has the exact same real optical imaging bias.  But we are at our core a real optical imaging project, and it seems reasonable to occasionally use shortcuts in nomenclature that reflect that.  It's also probably not reasonable to use such a shortcut in *every* context where we could.

In any case, if this is controversial, code review doesn't provide any way resolving that conflict; only RFC does.  And since "Wcs" is taken by an existing class right now, it's not going to be renamed to that in this ticket anyway, so it's not urgent.